server/lib/pbench/indexer.py

```diff
@@ -3204,6 +3204,11 @@ def mk_sosreports(tb, extracted_root, logger):
             logger.warning("Failed to fetch .md5 of sosreport {}: {}", sos, e)
             continue
         host_md = hostnames_if_ip_from_sosreport(os.path.join(extracted_root, sos))
+        if 'message' in host_md:
+            src = os.path.join(extracted_root, sos)
+            if os.path.exists(src):
+                os.rename(src,src+".BadHostname")
+            raise SosreportHostname(host_md['message'])
```

Comment:
Here, we'll check the status returned: if OK, we'll proceed as before and add the constructed dict to the sosreport list; if the returned status is an error, we'll munge the payload returned (the description of the problem) into a different dictionary (we'll have to decide the exact format) and *also* add the dictionary to the sosreportlist.

The next step is to investigate the *caller(s)* of mk_sosreports: they will get back a list of dicts but some of those will be from *failed* sosreports, so they will need to handle those dicts differently from the current normal handling. But the dicts (good or bad) will be indexed nevertheless, because we handle the bad cases as well as the good cases: we do not raise an exception the way we do today.

Reply:
BTW, we don't want to raise an exception in l3211: that would defeat the purpose. Also, renaming the file is ineffective, because it's a temporary directory: once the indexer is done, the temp directory is going to disappear without a trace.

Reply:
ok